### PR TITLE
Re-enable Lease Share

### PR DESCRIPTION
**Pull request by microzchang**
# Pull Request Checklist

Please leverage this checklist as a reminder to address commonly occurring feedback when submitting a pull request to make sure your PR can be reviewed quickly:

See the detailed list in the [contributing guide](https://github.com/Azure/azure-sdk-for-cpp/blob/main/CONTRIBUTING.md#pull-requests).

- [ ] [C++ Guidelines](https://azure.github.io/azure-sdk/cpp_introduction.html)
- [ ] Doxygen docs
- [ ] Unit tests
- [ ] No unwanted commits/changes
- [ ] Descriptive title/description
  - [ ] PR is single purpose
  - [ ] Related issue listed
- [ ] Comments in source
- [ ] No typos
- [ ] Update changelog
- [ ] Not work-in-progress
- [ ] External references or docs updated
- [ ] Self review of PR done
- [ ] Any breaking changes?

### CODE REVIEW

**Review by Jinming-Hu — APPROVED** [review]
LGTM

File: sdk/storage/azure-storage-files-shares/swagger/README.md [hunk 1]
```diff
@@ -969,6 +969,16 @@ directive:
...
 ```
 
+### BreakShareLease
+
+```yaml
+directive:
+  - from: swagger-document
+    where: $["x-ms-paths"]["/{shareName}?restype=share&comp=lease&break"].put.responses["202"].headers
+    transform: >
+      delete $["x-ms-lease-id"];
```
Comment (Jinming-Hu) [inline sdk/storage/azure-storage-files-shares/swagger/README.md:979]: Can you double-check this patch won't change the output of `rest_client.hpp`?

I didn't see the file change in this PR.

**Review by Jinming-Hu — APPROVED** [review]

File: sdk/storage/azure-storage-files-shares/swagger/README.md (hunk 1 above)
Comment (microzchang) [reply sdk/storage/azure-storage-files-shares/swagger/README.md:979]: You are right. I forgot to copy rest_client.hpp to repo and I fixed it now.

**Pull request merged**